### ISSUE TITLE
Explore generalisability of the generator functions

### ISSUE THREAD

**Issue by seabbs** [epinowcast/epidist#462]
I can see that the `log_lik` one has the name in. I don't think these functions will work by default with e.g. the marginal model. Is the point that they'll be modified then? Some question about whether for the time being for consistency they should be named `latent` given that they don't work with the marginal model as is (and the marginal model doesn't exist yet)?

_Originally posted by @athowes in https://github.com/epinowcast/epidist/pull/459#discussion_r1851801180_
            
Focus on if they can be used across a range of models (or what is needed to make this be the case). If they can be but only a subset explore how to make this clear to users

**Comment by athowes** [comment]
Interested in current status here. I can't tell if the generator functions are already working across different models. It looks like from the tests they do: https://github.com/epinowcast/epidist/blob/main/tests/testthat/test-gen.R (perhaps aside the naive model). So maybe this can be closed.

**Comment by seabbs** [comment]
Yeah they are so I think this can probably be closed if you are happy.
